lightly_studio/src/lightly_studio/core/dataset_query/dataset_query.py

```diff
@@ -20,7 +20,102 @@
 
 
 class DatasetQuery:
-    """Class for executing querying on a dataset."""
+    """Class for executing a query on a dataset.
+
+    # Filtering, ordering, and slicing samples in a dataset
+    Allows filtering, ordering, and slicing of samples in a dataset.
+    This class can be accessed via calling `.query()` on a Dataset instance.
+    ```python
+    dataset : Dataset = ...
+    query = dataset.query()
+    ```
+    The `match()`, `order_by()`, and `slice()` methods can be chained in this order.
+    You can also access the methods directly on the Dataset instance:
+    ```python
+    dataset.match(...) # shorthand for dataset.query().match(...)
+    ```
+
+    The query is lazily evaluated when iterating over it or converting it to a list.
```

Comment:
Minor: I really think we should mention the database:

```suggestion
    The object is converted to a SQL query that is lazily evaluated when iterating over it or converting it to a list.
```